mmstudio/src/main/java/org/micromanager/acquisition/internal/acqengjcompat/AcqEngJAdapter.java

```diff
@@ -448,14 +417,19 @@ private Iterator<AcquisitionEvent> createAcqEventIterator(SequenceSettings acqui
          zStack = MDAAcqEventModules.zStack(0,
                acquisitionSettings.slices().size() - 1,
                acquisitionSettings.sliceZStepUm(),
-               origin);
+               origin,
+               chSpecs);
       }
 
       if (acquisitionSettings.useChannels()) {
-         channels = MDAAcqEventModules.channels(acquisitionSettings.channels());
-         //TODO: keep shutter open
-         //TODO: skip frames
-         //TODO: z stack off for channel
+         if (chSpecs.size() > 0) {
+            Integer middleSliceIndex = (acquisitionSettings.slices().size() - 1) / 2;
+            channels = MDAAcqEventModules.channels(chSpecs, middleSliceIndex);
+         }
+         // TODO: keep shutter open, I believe this needs to be done in AcqEngJ; there should be
+         //    a flag in AcquisitionEvent that the Engine should look at, much like how it checks
+         //    if sequencing is possible.
+         // TODO: Check if z stack off for channel image is taken at correct Z
```

Comment:
AcqEngJ provides an API for this:
https://github.com/micro-manager/AcqEngJ/blob/3798f07bd0d1a59a81c7992d665d5ee2e9708ae6/src/main/java/org/micromanager/acqj/util/AcqEventModules.java#L140

Reply:
Oh nevermind I see that you have it below

Reply:
Oops, are we now adding an offset twice?